### PR TITLE
Dev/jsequeirag/vbs22 92 uso de marcadores

### PR DESCRIPTION
**Pull request by DieS06**
Location marker, location save as an on click function to endpoint post location, double map integrated in a single map component.

### CODE REVIEW

File: src/app/components/map/map.component.ts [hunk 1]
```diff
@@ -1,29 +1,126 @@
...
+
+              this.loc = {
+                address: this.address,
+                placeId: this.placeId,
+                latitude: this.latitude,
+                longitude: this.longitude
+              };
+
+            } else {
+              console.log("No results found");
```
Comment (nefi-cenfo) [inline src/app/components/map/map.component.ts:113]: Please remove console logs.

File: src/app/services/map-location.service.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        console.error('Error fetching locations', error);
+      }
+    });
+  };
+
+  saveLocation(location: ILocation): void {
+    this.add(location)
+    .subscribe({
+      next: (response) => {
+        //this.locationListSignal.update(locations => [response, ...locations])
```
Comment (nefi-cenfo) [inline src/app/services/map-location.service.ts:61]: Is this commented code needed?

File: src/app/services/map-location.service.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        console.log("Location saved successfully:", response);
```
Comment (nefi-cenfo) [inline src/app/services/map-location.service.ts:62]: Please remove console log.

File: src/app/services/map-location.service.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        console.error("Error saving location:", error);
+      }
+    });  
+  };
+
+  deleteLocation(location: ILocation): Observable<any> {
+    return this.logicDelete(location.id, location).pipe(
+      tap((response: any) => {
+        const deletedLocation = this.locationListSignal().map(loc => loc.id === location.id ? response : loc);
+        console.log(deletedLocation);
```
Comment (nefi-cenfo) [inline src/app/services/map-location.service.ts:74]: Please remove console log.

**Review by nefi-cenfo — APPROVED** [review]

File: src/app/services/map-location.service.ts
Comment (kendallduranv) [inline src/app/services/map-location.service.ts:1]: Maybe we can considerate in a future to add a  "try and catch" to manage the request sended

Comment (DieS06) [reply src/app/services/map-location.service.ts:1]: Please tag your comment to a piece of code.

File: src/app/interfaces/placeSearch.ts
Comment (kendallduranv) [inline src/app/interfaces/placeSearch.ts:1]: are all of them optional, if not you could use a default value at the service and put the ? just to the ones that are optional. You can look for an example to the flights service with onGetDefaultVoiageFlight and the line     this.flightSelected = this.flightService.onGetDefaultVoiageFlight(); at the constructor of the flight card component. Can watch them at the development branch

**Review by kendallduranv — DISMISSED** [review]
The branchs seems to be no up to date to the development one, so theres a posibility to have  conflicts.

**Review by ichivancrCenfotec — DISMISSED** [review]

File: src/app/interfaces/placeSearch.ts
Comment (DieS06) [reply src/app/interfaces/placeSearch.ts:1]: Pending for the merge of activities branch, to make some changes in location and places autocomplete.

**Review by kendallduranv — DISMISSED** [review]

**Review by nefi-cenfo — DISMISSED** [review]

**Pull request closed without merging**